### DISCUSSION TITLE
Feature Request: Sunrise/sunset based USB capture on/off

### DISCUSSION THREAD

**Discussion by pandel** [awawa-dev/HyperHDR#103] [Ideas]
Hi!

It would be really great, if we were able to setup a time range, in between HyperHDR simply won't start USB capturing.

My Ambilight setup is controlled via WLED and can be used as normal lighting for my living room, too. So it is never really disconnected from the power. Regarding Ambilight, I activate it only if it is dark outside, but at the moment, as soon as I turn on my TV, HyperHDR starts to light everything up, regardless if it is dark outside or not. So, i.e., if you could say, USB capturing off at 06:00 and on at 21:00 or so, one would not have to turn it off manually at times, when sunlight floods your living room :-). Even better, if those times could be set via API call, you could provide the time range for sunrise/sunset via home automation i.e. once a day, as they are changing over time.

Regards,
Holger

**Comment by awawa-dev** [comment]
Hi
Did you consider using Linux cron or Windows task scheduler and HyperHDR's API v17?   
  
![obraz](https://user-images.githubusercontent.com/69086569/130217133-ef9f6153-8e51-4790-b31f-925fe87f895b.png)  
  
That option disables/enables all instances and components...even USB grabber.  
The wizard will build two HTTP/S links for you (one: on, second: off), all is needed is to just invoke them from the scheduler.
For Linux there are cron tools like ````sunwait```` or ````heliocron```` with a functionality that you are looking for.  
Regards  
Awawa

**Reply by pandel** [reply]
Ahhhh! Many thanks! I am obviously still learning about all the features HyperHDR has to offer. I did not know about this. I will try to setup a corresponding script in my home automation system that is able to call the URL.

**Reply by awawa-dev** [reply]
You are excused: till v17 there was no JSON API documentation for HyperHDR in any form ;)
